extract-docker-tag/main.go

```diff
@@ -71,20 +72,55 @@ func extract(action *githubactions.Action) (result result, err error) {
 		branch := action.Getenv("GITHUB_HEAD_REF")
 		parts = strings.Split(strings.ToLower(branch), "/") // for branches like "dependabot/submodules/XXX"
 		result.tag = "pr-" + parts[len(parts)-1]
+
 	case "push", "schedule", "workflow_run":
-		branch := action.Getenv("GITHUB_REF_NAME")
-		if branch == "main" { // build on pull_request/pull_request_target for other branches
+		refName := action.Getenv("GITHUB_REF_NAME")
+		refType := action.Getenv("GITHUB_REF_TYPE")
+
+		result.tag, err = getTag(refName, refType)
+		if err != nil {
+			return
+		}
+
+		// build on pull_request/pull_request_target for other branches
+		switch refType {
+		case "branch":
+			result.name += "-dev"
+		case "tag":
 			result.name += "-dev"
-			result.tag = strings.ToLower(branch)
+		default:
+			panic("unreachable code")
 		}
+	default:
+		panic("unreachable code")
 	}
+
 	if result.tag == "" {
 		err = fmt.Errorf("failed to extract tag for event %q", event)
 		return
 	}
-
-	// set ghcr
 	result.ghcr = fmt.Sprintf("ghcr.io/%s/%s:%s", result.owner, result.name, result.tag)
 
 	return
 }
+
+func getTag(refName, refType string) (tag string, err error) {
```

Comment:
Let's not use named results if they don't add anything.
Let's also add a comment explaing what this function does.
And should it accept refType at all?

Reply:
> And should it accept refType at all?
otherwise  if's will be outside the function, I don't want this: now it's clear what happens with the tag value - it's in one place.

Reply:
> Let's also add a comment explaing what this function does.

@AlekSi comment added.